### DISCUSSION TITLE
Merge specific object from select

### DISCUSSION THREAD

**Discussion by p3lim** [mikefarah/yq#895] [Q&A] [answered]
`file1.yaml`:
```yaml
data:
  values: |
    foo: bar
```
`file2.yaml`:
```yaml
john: 42
```

I can merge `data.values` from the first file with the second file using yq _twice_:
```bash
yq e '.data.values' file1.yaml | yq ea 'select(fi==1) * select(fi==0)' file2.yaml -
````
Which outputs this:
```yaml
john: 42
foo: bar
```

But I'm wondering if this is possible with just calling yq once, e.g. something like this:
```bash
yq ea 'select(fi==0).data.values * select(fi==1)' file1.yaml file2.yaml
```

**Comment by mikefarah** [comment, marked as answer]
Yeah the issue is that data.values is actually a _string_ block, as denoted by the '|' - the contents of which happens to be a (now) string encoded yaml file.

Running once to extract and then again to process makes sense
